include/mpsl/mpsl_work.h

```diff
@@ -49,6 +49,32 @@ static inline int mpsl_work_submit(struct k_work *work)
 	return k_work_submit_to_queue(&mpsl_work_q, work);
 }
 
+/** @brief Submit an idle work item to the MPSL workqueue after a delay.
+ *
+ * @note Can be called by ISRs.
+ *
+ * @note
+ * Work items submitted to the MPSL workqueue should avoid using handlers
+ * that block or yield since this may prevent the MPSL workqueue from
+ * processing other work items in a timely manner.
+ *
+ * @note This is a no-op if the work item is already scheduled or submitted,
+ * even if @p delay is @c K_NO_WAIT.
+ *
+ * @param dwork Address of delayable work item.
+ *
+ * @param delay the time to wait before submitting the work item.  If @c
+ * K_NO_WAIT and the work is not pending this is equivalent to
+ * mpsl_work_submit().
+ *
+ * @return N/A
```

Comment:
I'm not sure why the existing function had this. If the return value from `k_work_*` is negative, we should assert. The Zephyr docs make a point about the importance of checking the return values: https://docs.zephyrproject.org/latest/kernel/services/threads/workqueue.html#check-return-values.